src/components/LoginDialog.jsx

```diff
@@ -1,32 +1,36 @@
 import React from 'react';
+import { useLocation, useHistory } from 'react-router-dom';
 import Dialog from '@material-ui/core/Dialog';
 import DialogContent from '@material-ui/core/DialogContent';
 import DialogContentText from '@material-ui/core/DialogContentText';
 import DialogTitle from '@material-ui/core/DialogTitle';
-
 import firebase from 'firebase';
 import StyleFirebaseAuth from 'react-firebaseui/StyledFirebaseAuth';
-import PropTypes from 'prop-types';
 
-const uiConfig = {
-  signInFlow: 'popup',
-  signInOptions: [
-    firebase.auth.GoogleAuthProvider.PROVIDER_ID,
-    firebase.auth.EmailAuthProvider.PROVIDER_ID,
-    firebase.auth.FacebookAuthProvider.PROVIDER_ID,
-  ],
-  credentialHelper: 'none', // hacky way to disable redirect on email login
-  callbacks: {
-    signInSuccessWithAuthResult: () => false,
-  },
-};
+export default function LoginDialog() {
+  const location = useLocation();
+  const { showLoginDialog } = location.state || { showLoginDialog: false };
+  const history = useHistory();
+
+  const closeDialog = () =>
+    history.replace({ state: { showLoginDialog: false } });
```

Comment:
I suggest ordering/grouping these so that related statements are together, e.g.
```js
  const location = useLocation();
  const { showLoginDialog } = location.state || { showLoginDialog: false };
  
  const history = useHistory();
  const closeDialog = () =>
    history.replace({ state: { showLoginDialog: false } });

```

The extra blank line at the bottom is so that `uiConfig` is separated from relatively unrelated lines.